docs/virtual_machine_template_guide.adoc

```diff
@@ -0,0 +1,189 @@
+= VirtualMachine Templates
+
+VirtualMachine templates are openshift templates (https://docs.openshift.org/latest/dev_guide/templates.html) +
+that describe VirtualMachine specifications which produce the following template objects: +
+**One** OfflineVirtualMachine and PersistentVolumeClaim. +
+The following document describes the process of creating a Virtual machine template which will be managed by ManageIQ.
+
+== Uploading a Template
+
+If you have a YAML file that defines a VirtualMachine template, you can upload the template to an Openshift cluster using the CLI. 
+To upload a template to your current project’s template library, pass the YAML file with the following command: +
+
+----
+$ oc create -f <filename>
+----
+
+== Modifying an Uploaded Template
+You can edit a template that has already been uploaded to your project by using the following command: +
+
+----
+$ oc edit template <template>
+----
+
+Writing Templates
+You can define new VirtualMachine templates to make it easy to recreate all relevant objects (offlineVirtualMachine and PersistentVolumeClaim). The template will define the objects it creates along with some metadata to guide the creation of those objects.
+
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha1
+kind: Template
+metadata:
+  name: RHEL-7 ❶
+objects:
+- apiVersion: kubevirt.io/v1alpha1
+  kind: offlineVirtualMachine ❷
+  metadata:
+    name: rhel-7-${NAME} ❸
+  spec:
+    template:
+      metadata:
+      labels:
+        my: label
+      spec:
+        domain:
+          resources: ❹
+            memory: ${MEMORY} 
+            cpu: ${CPU}
+          devices:
+      disks: ➎
+        name: disk0
+        volumeName: mypvc 
+      volumes: ➏
+         name: mypvc
+         presistentVolumeClaim:
+      claimName: rhel-7-pvc-${NAME}
+- apiVersion: kubevirt.io/v1alpha1
+  kind: presistentVolumeClaim ❼
+  metadata:
+    name: rhel-7-pvc-${NAME} ❽
+labels: ...       
+parameters: ...
+
+----
+ 
+❶ The unique name of the template. +
+
+❷ The OfflineVirtualMachine Object. This object must appear only once in the template. +
+
+❸ OfflineVirtualMachine’s name. +
+
+❹ OfflineVirtualMachine’s resources, currently includes only memory and cpu. +
+   Cpu is the number of cpu cores the Virtual Machine requires. +
+   Memory is the minimal amount the Virtual Machine requires. +
```

Comment:
Can we add information how units are encoded?

Reply:
@yanirq apart from examples in the k8s docs, here is the best description i know: https://github.com/kubernetes/kubernetes/blob/a523084d88edc929d817f828d66b3091350d2f17/staging/src/k8s.io/apimachinery/pkg/api/resource/quantity.go#L35

Reply:
added a link

Reply:
@rmohr do you have kubevirt docs describing that we support quantities?

IOW Is there a PR describing the VM API design approach?